### ISSUE TITLE
add tile providers for OpenStreetMap and ESRI Imagery

### ISSUE THREAD

**Issue by comperem** [bokeh/bokeh#9560]
Hello,

**TL;DR**
This issue is to initiate a pull request to add the following tile providers to the Bokeh file **`./bokeh/tile_providers.py`:**
- OpenStreetMaps map tile provider
- Wikimedia / Wikipedia map tile provider (via OSM)
- ESRI Imagery with satellite imagery

**Background:**
In my search for a Google Maps replacement, the current Bokeh has tile providers are a good start but the Stamen and Carto tiles do not provide a combination of:
- ability for close zoom-in (map data not available)
- detail at high zoom (it's just blank, or not descriptive)
- satellite imagery

**Licensing:**
The [OpenStreetMap data](www.openstreetmap.org) is an excellent map-style alternative and the ESRI Imagery tiles are excellent satellite image replacements. OSM is entirely open-source. Esri has [generous open-source](https://www.esri.com/en-us/arcgis/open-vision/standards/open-source) and [open-data policies](https://www.esri.com/en-us/arcgis/open-vision/standards/open-data). Neither of these require api keys or registration, although responsible usage is, indeed, encouraged and enforced. Unusually high usage places a penalty on volunteer and donated hardware and network services.

But, for simple basemap overlays (underlays), prototyping for research purposes, or usage in other open-source projects, these two are outstanding.

**Rationale:**
Both of these tile providers, and others, are already integrated into [PyViz](http://pyviz.org),  [Holoviews](http://holoviews.org/), and [GeoViews](http://geoviews.org/), all of which rely on Bokeh as an underlying library. Rather than adding layers (and moving to Anaconda package management) to access those map tile providers, it seems Bokeh could (should) have these maps available directly.

**Example:**
The screenshot at this link shows multiple failed or poorly displaying zoomed-in options from the fine work presented at the [GeoViews page on working with Bokeh](http://geoviews.org/user_guide/Working_with_Bokeh.html):

https://discourse.bokeh.org/t/open-source-tile-provider-replacement-for-google-maps/4462

Marc Compere

**Comment by bryevdv** [comment]
@comperem This seems like a great addition, then. Do you think you could submit the PR to add new instances to `bokeh.tile_providers`?

**Comment by comperem** [comment]
Umm, yes, possibly. I'd like to but the PR process is not obvious. It'll take some reading on my part. The changes are pretty straightforward to a single file (`./bokeh/tile_providers.py`) but the PR is new for me.

**Comment by bryevdv** [comment]
@comperem Almost there :) Instead of cloning the main repo (which you will not have permissions to push to) you need to first *fork* the repo in to your own GH account. If you look in the very upper right of the page there is a button to fork the repo. Once this is done, clone your fork, and push the change there (you will have permission to since the fork is in your account). Finally, on the main page:

https://github.com/bokeh/bokeh

There is a "New Pull Request" button where you can specify that you would like for us to pull changed from your fork into the master branch of this main repo. Let me know if I can elaborate on anything!
